podpac/core/cache/cache.py

```diff
@@ -795,16 +607,45 @@ def get(self, node, key, coordinates=None):
         CacheError
             If the data is not in the cache.
         '''
-        listing = CacheListing(node=node, key=key, coordinates=coordinates)
-        paths = self.cache_glob(node, key, coordinates)
-        for p in paths:
-            c = self.load_container(p)
-            if c.has(listing):
-                data = c.get(listing).data
-                if data is None:
-                    CacheException("Stored data is None.")
-                return data
-        raise CacheException("Cache miss. Requested data not found.")
+
+        path = self.find(node, key, coordinates)
+
+        if path is None:
+            raise CacheException("Cache miss. Requested data not found.")
+
+        # read
+        s = self._load(path)
+        
+        # deserialize
+        if path.endswith('.netcdf'):
+            data = xr.open_dataarray(s)
```

Comment:
This `data` should be fed to a `UnitsDataArray` because we have some special deserialization that happens for units that may be part of the metadata.